### PR TITLE
Add Snackbar for Offline

### PR DESCRIPTION
**Pull request by sachincool**
## Description
Added Snackbars for Offline (No Internet) When there is no Internet connectivity.

## Related issues and discussion
Fixes #762 
 
 ## Screen-shots, if any
 
 ## Checklist
 
Make sure you've done all the following (You can delete the checklist before submitting)
 
 - [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.
 - [x] Code is well documented
 - [ ] Include unit tests for new functionality
 - [x] All user-visible strings are made translatable
 - [x] Code passes Travis builds in your branch
 - [x] If you have multiple commits please combine them into one commit by squashing them.
 - [x] Read and understood the contribution guidelines .

### CODE REVIEW

**Comment by sachincool** [comment]
Note For The Reviewers : 
* I didn't change inside the History. 
* If you go back from a fragment like Search Barcode Back to MainActivity, due to the `launchMode: SingleTop` the snackbar instance isn't recreated. hence you only see the snackbar once. I was doubtful whether it is good like this or should we let the snackbar stay there throughout the whole user experience. 
* as per @teolemon suggested i thought using snackbars was more _cool_ :smile:  

@policorn  Kindly Close https://github.com/openfoodfacts/openfoodfacts-androidapp/pull/762#issuecomment-368646969

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/CustomTextView.java [hunk 1]
```diff
@@ -28,6 +31,16 @@ public CustomTextView(Context context, AttributeSet attrs, int defStyleAttr) {
         initAttrs(context, attrs);
     }
 
+    public static void CustomOfflineSnackbar(View view) {
+
+        Snackbar.make(view, "Offline Mode (No Internet)", Snackbar.LENGTH_INDEFINITE)
```
Comment (policorn) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/CustomTextView.java]: Please make it translatable

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/WelcomeActivity.java [hunk 2]
```diff
@@ -28,6 +31,31 @@
...
+        public void onPageScrolled(int arg0, float arg1, int arg2) {
+
+        }
+
+        @Override
+        public void onPageScrollStateChanged(int arg0) {
+
+        }
+    };
     private PrefManager prefManager;
```
Comment (policorn) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/WelcomeActivity.java]: Please move it up with the rest of privates vars.

**Comment by sachincool** [comment]
@policorn  done :+1:

**Comment by Karljoones** [comment]
Hey @sachincool and @policorn, this pull request is the same as #856 that we closed off after deciding that this feature was not necessary for the app. Can we close this too?

**Comment by sachincool** [comment]
@Karljoones cool. I have no issues with this one closing if this is not necessary

**Comment by Karljoones** [comment]
@sachincool Thank you for the work and sorry that we are not merging it!

**Pull request closed without merging**